docs/tests.md

```diff
@@ -0,0 +1,439 @@
+  * amount of desired machines might be reduced from 3 to 2 for each machineset
+
+Within this test machineset with 1 desired replica and with a specific set of labels is created.
+After, a Job with 6 desired replicas is attempting to be scheduled on nodes with labels configured for previously created machinesets.
+
+The test is expecting both machinesets to be scaled up to 3 machines and accept the payload. Then, created Job is removing and the machineset is expected to be scaled down back to 1 replica.
+
+After machinesets will be scaled back to 1 replica test checks if no cluster-autoscaler specific annotations are presented on machines and no autoscaler specific taints are presented on respective nodes from these machinesets.
+
+Test considered failed if scale up or down did not happen, or after scaling down there are taints/annotations leftovers within a 3 minutes timeout.
+
+#### Autoscaler should use a ClusterAutoscaler that has 12 maximum total nodes count and balance similar nodes enabled scales up and down while respecting MaxNodesTotal [Slow][Serial]
+* Parallel: No
+* ExecTime: ~13m on AWS
+* Recommendation:
+  * Since this test does not leverage or checking `balance similar nodes` feature, this might be extracted to separate group with reducing `MaxNodesTotal` number to ~8. This should speed up this test a bit and reduce resource consumption.
+  * This test requires revise and extension of checks. Workload destiny need to be checked too. As i understand this test at the moment of writing, part of the workload should not be scheduled due to `MaxNodesTotal` limit.
+  * Gather and report a number of nodes when the test starts
+  * Might be merged with from/to zero scaling test ([Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count It scales from/to zero](#autoscaler-should-use-a-clusterautoscaler-that-has-100-maximum-total-nodes-count-it-scales-fromto-zero)).
+
+This test excersising `MaxNodesTotal` limit for the autoscaler. Intially it creates machineset with 1 desired replica and a Job
+with `MaxNodesTotal` + 1 desired pods in it. Then waits cluster to be scaled up till `MaxNodesTotal` nodes number and checks that this constraint was not violated.
+
+Then workload deletes and test waits for the machineset to be scaled back to 1 replica.
+
+#### Autoscaler should use a ClusterAutoscaler that has 12 maximum total nodes count and balance similar nodes enabled places nodes evenly across node groups [Slow]
+* Parallel: No
+* ExecTime: ~10m on AWS
+* Recommendation:
+  * Check kubernetes-autoscaler deployment for the `balance-similar-node-groups` flag presence first
+  * Currently max machineset size for machine autoscaler resource is 3 within this test. I suggest to increase it to 4 or 6 for being more sure that `balance-similar-node-groups` does its thing and that the test pass is not a coincidence.
```

Comment:
+1, also as discussed in standup, we might want to create another machineset, with zero replicas, that does not match the job to ensure that we aren't placing nodes in the empty machineset.